src/main/java/ch/njol/skript/registrations/Classes.java

```diff
@@ -319,6 +320,18 @@ public static <T> ClassInfo<? super T> getSuperClassInfo(final Class<T> c) {
 		return null;
 	}
 
+	/**
+	 * Gets the class info of the super type of given classes or its closest registered superclass.
+	 * This method is useful for Skript to avoid passing around "unknown" super types.
+	 *
+	 * @param classes The classes to determine a super type from.
+	 * @return The closest info for the super type of <code>classes</code>.
+	 */
+	// This method is used to avoid issues like https://github.com/SkriptLang/Skript/issues/5848
+	public static ClassInfo<?> getSuperClassInfo(Class<?>... classes) {
+		return getSuperClassInfo(Utils.getSuperType(classes));
```

Comment:
Is this not a recursion?

Reply:
This method is used:
https://github.com/SkriptLang/Skript/blob/4fc171fd8eb6ccb97cb0fccec5c94a1d6ef1c99e/src/main/java/ch/njol/skript/registrations/Classes.java#L306